app/routes.py

```diff
@@ -1,2 +1,180 @@
+
+def slack_post_message(title):
+
+    url = 'https://slack.com/api/chat.postMessage'
+    params = {
+        "channel":"task-notification",
+        "text":f"Someone just completed the task {title}"
+    }
+    header={
+        "Authorization": f"Bearer {os.environ.get('API_TOKEN')}" }
+
+    req = requests.post(url, params=params, headers=header)
+    r = req.json()
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"], strict_slashes=False)
+def get_single_task(task_id):
+
+    task = Task.query.get(task_id)
+    if task is None:
+        return jsonify(None),404
+
+    if request.method == "GET":
+        return {"task": task.get_resp()}, 200
+
+    elif request.method == "PUT":
+        form_data = request.get_json()
+        task.title = form_data["title"]
+        task.description = form_data["description"]
+        task.completed_at = form_data["completed_at"]
+        db.session.commit()
+        return {"task": task.get_resp()}, 200
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return {
+            "details":f'Task {task.task_id} "{task.title}" successfully deleted'
+        }, 200
+
+@tasks_bp.route("", methods=["GET", "POST"])
+def handle_tasks():
+    if request.method == "GET":
+        sort_query = request.args.get("sort")
+        
+        if sort_query == 'asc':
+            tasks = Task.query.order_by(Task.title.asc())
+        elif sort_query == 'desc':
+            tasks = Task.query.order_by(Task.title.desc())
+        else:
+            tasks = Task.query.all()
+        
+        task_response = []
+        for task in tasks:
+            task_response.append(task.get_resp())
+        return jsonify(task_response), 200
+    
+    elif request.method == "POST":
+        request_body = request.get_json()
+        if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+            return {"details":"Invalid data"}, 400
+
+        else:
+            new_task = Task(title=request_body["title"],
+                        description=request_body["description"],
+                        completed_at=request_body["completed_at"])
+
+        db.session.add(new_task)
+        db.session.commit()
+        return {"task": new_task.get_resp()}, 201
+
+@goals_bp.route("", methods=["GET","POST"])
+def handle_goals():
+    if request.method == "GET":
+        goals = Goal.query.all()
+        goal_response = []
+        for goal in goals:
+            goal_response.append(goal.get_resp())
+        return jsonify(goal_response), 200
+
+    elif request.method == "POST":    
+        request_body = request.get_json()
+        if "title" not in request_body:
+            return {"details":"Invalid data"}, 400
+        else:
+            new_goal = Goal(title=request_body["title"])
+
+            db.session.add(new_goal)
+            db.session.commit()
+            return {"goal": new_goal.get_resp()}, 201
```

Comment:
Nothing to add, this function looks great and so neat 😍 ! If you'd like to refactor in the future I'd suggest trying to create `goal_response` using list comprehension 👀